services/web/server/tests/unit/with_dbs/03/test_redis.py

```diff
@@ -44,13 +76,44 @@ async def test_redlocks_features(redis_client: aioredis.Redis):
                 ...
     assert not await lock.locked()
 
-    # now create a lock with a ttl
-    ttl_lock = redis_client.lock("ttl_resource", timeout=2, blocking_timeout=1)
+
+@pytest.fixture
+def lock_timeout() -> datetime.timedelta:
+    return datetime.timedelta(seconds=2)
+
+
+async def test_lock_with_ttl(
+    redis_client: aioredis.Redis, faker: Faker, lock_timeout: datetime.timedelta
+):
+    ttl_lock = redis_client.lock(faker.pystr(), timeout=lock_timeout.total_seconds())
     assert not await ttl_lock.locked()
+
     with pytest.raises(redis.exceptions.LockNotOwnedError):
         # this raises as the lock is lost
         async with ttl_lock:
             assert await ttl_lock.locked()
             assert await ttl_lock.owned()
-            await asyncio.sleep(3)
+            await asyncio.sleep(2 * lock_timeout.total_seconds())
             assert not await ttl_lock.locked()
+
+
+async def test_lock_with_auto_extent(
+    redis_client: aioredis.Redis, faker: Faker, lock_timeout: datetime.timedelta
+):
+    ttl_lock = redis_client.lock(faker.pystr(), timeout=lock_timeout.total_seconds())
+    assert not await ttl_lock.locked()
+
+    async def _auto_extend_lock(lock: Lock) -> None:
+        assert await lock.reacquire() is True
+
+    async with ttl_lock, periodic_task(
+        _auto_extend_lock,
+        interval=0.6 * lock_timeout,
+        task_name=f"{ttl_lock.name}_auto_extend",
+        lock=ttl_lock,
+    ):
+        assert await ttl_lock.locked()
+        assert await ttl_lock.owned()
+        await asyncio.sleep(5 * lock_timeout.total_seconds())
+        assert await ttl_lock.locked()
+        assert await ttl_lock.owned()
```

Comment:
Question: Would it make sense to also test for the lock being released when exiting the context manager?